wgsl/index.bs

```diff
@@ -9522,18 +9523,24 @@ The <dfn export>interpolation sampling</dfn> [=shader-creation error|must=] be o
 :: Interpolation is performed per sample.
     The [=fragment=] shader is invoked once per sample when this attribute is
     applied.
+: <dfn for="interpolation sampling" export>first</dfn>
+:: The interpolated value is constant and provided by the first vertex of the primitive.
+: <dfn for="interpolation sampling" export>eitehr</dfn>
+:: The interpolated value is constant and provided by the first vertex or last vertex of the primitive.
+    Whether the value comes from the first or last vertex is implementation dependent.
 
 For user-defined IO of scalar or vector floating-point type:
 * If the interpolation attribute is not specified, then `@interpolate(perspective, center)` is assumed.
 * If the interpolation attribute is specified with an interpolation type:
-    * If the interpolation type is `flat`, then interpolation sampling [=shader-creation error|must not=] be specified.
+    * If the interpolation type is `flat`, then:
+        * interpolation sampling [=shader-creation error|must=] be `first`, `either`, or not be specified.
+        * If interpolation sampling not specified, `first` is assumed.
     * If the interpolation type is `perspective` or `linear`, then:
-         * Any interpolation sampling is valid.
-         * If interpolation sampling is not specified, `center` is assumed.
+        * interpolation sampling [=shader-creation error|must=] be `center`, `centroid`, `sample`, or not be specified.
+        * If interpolation sampling is not specified, `center` is assumed.
```

Comment:
```suggestion
        * If interpolation sampling is specified, it [=shader-creation error|must=] be `center`, `centroid`, or `sample`.
        * Otherwise, `center` is assumed.
```